tests/cases/resample.py

```diff
@@ -0,0 +1,107 @@
+            ArrayKeys.RAW,
+            ArraySpec(
+                roi=Roi((0, 0, 0), (1000, 1000, 1000)),
+                voxel_size=(4, 4, 4)))
+
+        self.provides(
+            ArrayKeys.GT_LABELS,
+            ArraySpec(
+                roi=Roi((0, 0, 0), (1000, 1000, 1000)),
+                voxel_size=(4, 4, 4)))
+
+    def provide(self, request):
+
+        batch = Batch()
+
+        # have the pixels encode their position
+        for (array_key, spec) in request.array_specs.items():
+
+            roi = spec.roi
+
+            for d in range(3):
+                assert roi.get_begin()[d]%4 == 0, "roi %s does not align with voxels"
+
+            data_roi = roi/4
+
+            # the z,y,x coordinates of the ROI
+            meshgrids = np.meshgrid(
+                    range(data_roi.get_begin()[0], data_roi.get_end()[0]),
+                    range(data_roi.get_begin()[1], data_roi.get_end()[1]),
+                    range(data_roi.get_begin()[2], data_roi.get_end()[2]), indexing='ij')
+            data = meshgrids[0] + meshgrids[1] + meshgrids[2]
+
+            spec = self.spec[array_key].copy()
+            spec.roi = roi
+            batch.arrays[array_key] = Array(
+                    data,
+                    spec)
+        return batch
+
+class TestResample(ProviderTest):
+
+    def test_output(self):
+
+        ArrayKey('RAW_RESAMPLED')
+        ArrayKey('GT_LABELS_RESAMPLED')
+
+        request = BatchRequest()
+        request.add(ArrayKeys.RAW, (200,200,200))
+        request.add(ArrayKeys.RAW_RESAMPLED, (120,120,120))
+        request.add(ArrayKeys.GT_LABELS, (200,200,200))
+        request.add(ArrayKeys.GT_LABELS_RESAMPLED, (200,200,200))
+
+        pipeline = (
+                ResampleTestSource() +
+                Resample(ArrayKeys.RAW, Coordinate((8,8,8)), ArrayKeys.RAW_RESAMPLED) + #Test downsampling
+                Resample(ArrayKeys.GT_LABELS, Coordinate((2,2,2)), ArrayKeys.GT_LABELS_RESAMPLED, interp_order=0) #Test upsampling, without interpolation
+        )
+
+        with build(pipeline):
+            batch = pipeline.request_batch(request)
+
+        for (array_key, array) in batch.arrays.items():
+
+            # assert that pixels encode their position for supposedly unaltered 
+            # arrays
+            if array_key in [ArrayKeys.RAW, ArrayKeys.GT_LABELS]:
+
+                # the z,y,x coordinates of the ROI
+                roi = array.spec.roi/4
+                meshgrids = np.meshgrid(
+                        range(roi.get_begin()[0], roi.get_end()[0]),
+                        range(roi.get_begin()[1], roi.get_end()[1]),
+                        range(roi.get_begin()[2], roi.get_end()[2]), indexing='ij')
+                data = meshgrids[0] + meshgrids[1] + meshgrids[2]
+
+                self.assertTrue(np.array_equal(array.data, data), str(array_key))
+
+            elif array_key == ArrayKeys.RAW_RESAMPLED:
+
+                self.assertTrue(array.data[0,0,0] == 31, 
```

Comment:
Not sure if I follow why the expected value is 31 instead of 30. Since you request array of shape (120,120,120), and a total request size of (200,200,200) I would expect this array to be offset by 40 on each axis. voxel-size of 4 means an offset of 10 on each axis. Adding those up I would expect a value of 30.

Reply:
Yup. ["Behavior is not a pixel-perfect copy of down/upsample nodes, because this node relies on skimage.transform.rescale to perform non-integer scaling factors."](https://github.com/funkey/gunpowder/blob/445d2ed875e2d33de28ff786c9a9767ea84049f9/gunpowder/nodes/resample.py#L14) I believe the skimage function differs in how it localizes data, noting "that 0th pixel is at position (0.5, ...)" 
See [the source for reference.](https://github.com/scikit-image/scikit-image/blob/55dee6421c1411388eb299f70244c90d956140ef/skimage/transform/_warps.py#L224)

Please let me know if you think this is a significant issue that needs addressing, and if you have any suggestions for elegant solutions. Thank you!

Reply:
Oh nvm, it makes sense to me know. its averaging over the voxels in the raw roi: (40:48, 40:48, 40:48). Those values are (30,31,31,32,31,32,32,33). Average to 31.5, and casting to an int is rounding down for some reason. Wierd but ok. Maybe comment how you get to 31 for people who might come across this in the future and be confused like me.

Reply:
Commented in commit f2310f2.